std/prelude.asm

```diff
@@ -18,10 +18,28 @@ enum Option<T> {
 enum Constr {
     /// A polynomial identity, result of the "=" operator.
     Identity(expr, expr),
+
     /// A lookup constraint with selectors, result of the "in" operator.
     Lookup((Option<expr>, Option<expr>), (expr, expr)[]),
+
+    /// A "phantom" lookup constraint, i.e., an annotation for witness generation.
+    /// The actual constraint should be enforced via other constraints.
+    /// Contains:
+    /// - The selectors (if any) for the left and right hand side.
+    /// - The LHS and RHS values.
+    /// - The multiplicity column.
+    PhantomLookup((Option<expr>, Option<expr>), (expr, expr)[], expr),
```

Comment:
Nice, would this be useful for using stwo lookup? 

other questions: 
The left selector is used for the case where not all the values in LHS belongs to RHS? 
the right selector is used for the case where not all the RHS values are used? if it is, the multiplicity actually do the same thing?

Reply:
It's useful to get witness generation working when using our PIL implementation of lookup, which I think could be used with Stwo :)

> The left selector is used for the case where not all the values in LHS belongs to RHS?
> the right selector is used for the case where not all the RHS values are used?

Yes!

> if it is, the multiplicity actually do the same thing?

Kind of, the multiplicity with which elements are added to the bus will be `rhs_selector * multiplicity`. I guess actually it would be better (because of a lower constraint degree) to add a constraint like `rhs_selector * multiplicity = 0` and only use `multiplicity`.

While `multiplicity` is a witness column, `rhs_selector` is typically a fixes column (e.g. because a "block machine" only sends tuples to the bus every 4 rows).